### DISCUSSION TITLE
Connection using sqlalchemy and thin driver on M1 Mac

### DISCUSSION THREAD

**Discussion by bassg0navy** [oracle/python-oracledb#109] [Q&A] [answered]
On an M1 Mac and seeing the following error during import of Flask app: 

`File "/app/proxy-webserver/proxy/models.py", line 38, in <module>
proxy-webserver    |     engine = create_engine("oracle+oracledb://", creator=pool.acquire, poolclass=NullPool)
proxy-webserver    |   File "<string>", line 2, in create_engine
proxy-webserver    |   File "/usr/local/lib/python3.9/site-packages/sqlalchemy/util/deprecations.py", line 309, in warned
proxy-webserver    |     return fn(*args, **kwargs)
proxy-webserver    |   File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/create.py", line 534, in create_engine
proxy-webserver    |     entrypoint = u._get_entrypoint()
proxy-webserver    |   File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/url.py", line 655, in _get_entrypoint
proxy-webserver    |     cls = registry.load(name)
proxy-webserver    |   File "/usr/local/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 343, in load
proxy-webserver    |     raise exc.NoSuchModuleError(
proxy-webserver    | sqlalchemy.exc.NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:oracle.oracledb
`
Code to make connection in models.py below:

`engine = create_engine("oracle+oracledb://", creator=pool.acquire, poolclass=NullPool)
`

If I remove oracledb and leave just oracle in first part of string, I get similar error for cx_Oracle "dialect". This works fine on Windows 10. Due to issues with M1 chips, simply installing and importing cx_Oracle has not been a resolution for me. Also, app is running in container so it is ideal to use thin driver.

I can provide more details if necessary; decided to come here first before opening issue after unsuccessful search for similar discussions.

**Comment by cjbj** [comment, marked as answer]
Check https://levelup.gitconnected.com/using-python-oracledb-1-0-with-sqlalchemy-pandas-django-and-flask-5d84e910cb19 and https://medium.com/oracledevs/using-the-development-branch-of-sqlalchemy-2-0-with-python-oracledb-d6e89090899c
Also see the SQLAlchemy examples in https://github.com/cjbj/python-oracledb-demos-2022

If these don't help, post the exact versions of the modules you are using, and also a _complete_, runnable example.  Thanks!

**Comment by bassg0navy** [comment]
Thanks @cjbj 👍  Adding the following was exactly what I needed: 

`oracledb.version = "8.3.0"
sys.modules["cx_Oracle"] = oracledb`

Upgrading to sqlalchemy 2 is now on my to-do list.
